requests/resolver_cache.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2021 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package requests
+
+import "sync"
+
+var defaultResolveCache = NewResolverCache()
+
+type resolverCache struct {
+	mu    sync.Mutex
```

Comment:
We could consider using a RWMutex to allow for reading from the cache simultaneously. Do you think it could be an improvement?

Reply:
This is a very basic implementation. This is one of the improvements that will be applied later.